File: src/types/payment.ts
/**
 * Payment Types (Renderer Process)
 */

export enum PaymentStatus {
  Waiting = 'waiting',
  Confirming = 'confirming',
  Confirmed = 'confirmed',
  Sending = 'sending',
  PartiallyPaid = 'partially_paid',
  Finished = 'finished',
  Failed = 'failed',
  Refunded = 'refunded',
  Expired = 'expired',
}

export enum CreditPlan {
  Starter = 'starter',
  Pro = 'pro',
  Enterprise = 'enterprise',
}

export interface CreditPlanInfo {
  plan: CreditPlan;
  credits: number;
  priceUsd: number;
  popular?: boolean;
  description?: string;
}

export interface CreatePaymentRequest {
  plan: CreditPlan;
  pay_currency?: string;
}

export interface CreatePaymentResponse {
  payment_id: string;
  payment_status: PaymentStatus;
  pay_address: string;
  pay_amount: number;
  pay_currency: string;
  price_amount: number;
  price_currency: string;
  order_id: string;
  order_description: string;
  payment_url: string;
  created_at: string;
  updated_at: string;
  expiration_estimate_date: string;
}

export interface PaymentStatusResponse {
  payment_id: string;
  order_id: string;
  payment_status: PaymentStatus;
  pay_address: string;
  pay_amount: number;
  pay_currency: string;
  price_amount: number;
  price_currency: string;
  actually_paid: number | null;
  outcome_amount: number | null;
  created_at: string | null;
  updated_at: string | null;
}

export interface AvailableCurrency {
  code: string;
  name: string;
  logo_url: string;
}

export interface PaymentHistory {
  payment_id: string | null;
  order_id: string;
  plan: CreditPlan;
  credits_amount: number;
  price_amount: number;
  pay_address: string | null;
  pay_amount: number | null;
  pay_currency: string | null;
  status: PaymentStatus;
  credits_applied: boolean;
  created_at: string;
  updated_at: string;
}

File: src-tauri/src/types/payment.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreditPlan {
    pub id: String,
    pub name: String,
    pub credits: f64,
    pub price: f64,
    pub currency: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreditPlanInfo {
    pub plans: Vec<CreditPlan>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct AvailableCurrency {
    pub code: String,
    pub name: String,
    pub symbol: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreatePaymentRequest {
    pub plan_id: String,
    pub currency: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreatePaymentResponse {
    pub payment_id: String,
    pub payment_url: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct PaymentStatusResponse {
    pub status: String,
    pub credits: Option<f64>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct PaymentHistory {
    pub payments: Vec<PaymentRecord>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct PaymentRecord {
    pub id: String,
    pub amount: f64,
    pub currency: String,
    pub status: String,
    pub created_at: String,
    pub credits: f64,
}
